splash-dao-offchain/src/routines/inflation/actions.rs

```diff
@@ -67,39 +89,51 @@ pub trait InflationActions<Bearer> {
         &self,
         inflation_box: Bundled<InflationBoxSnapshot, Bearer>,
         factory: Bundled<PollFactorySnapshot, Bearer>,
+        current_slot: Slot,
+        funding_boxes: AvailableFundingBoxes,
     ) -> (
         SignedTxBuilder,
         Traced<Predicted<Bundled<InflationBoxSnapshot, Bearer>>>,
         Traced<Predicted<Bundled<PollFactorySnapshot, Bearer>>>,
         Traced<Predicted<Bundled<WeightingPollSnapshot, Bearer>>>,
+        FundingBoxChanges,
     );
     async fn eliminate_wpoll(
         &self,
         weighting_poll: Bundled<WeightingPollSnapshot, Bearer>,
-    ) -> SignedTxBuilder;
+        funding_boxes: AvailableFundingBoxes,
+        current_slot: Slot,
+    ) -> (SignedTxBuilder, FundingBoxChanges);
     async fn execute_order(
         &self,
         weighting_poll: Bundled<WeightingPollSnapshot, Bearer>,
         order: (VotingOrder, Bundled<VotingEscrowSnapshot, Bearer>),
-    ) -> (
+        current_slot: Slot,
+    ) -> Option<(
```

Comment:
To encode the possibility that voting order is defective. Can remove `Option` here and add validation of the order in `Behaviour::try_apply_votes()`. Need to validate thoroughly, otherwise user can submit a failing order, leading to script error and forfeiting of the bot's collateral.

Reply:
Is it possible to validate orders in advance right after the program received them from outer world?

Reply:
Yes, we can. I plan to do this in a subsequent PR where I can also incorporate the new `weighting_witness` script.

Reply:
Ideally in an instance responsible for "reading" it like here: https://github.com/splashprotocol/splash-offchain-multiplatform/blob/785bcf2ba4ca899945b933708aa185fa442db44c/bloom-offchain-cardano/src/orders/limit.rs#L419

Reply:
Ok, then I merge this one.